src/test/java/com/hm/achievement/db/SQLiteDatabaseNullSafetyTest.java

```diff
@@ -0,0 +1,117 @@
+package com.hm.achievement.db;
+
+import com.hm.achievement.AdvancedAchievements;
+import com.hm.achievement.exception.PluginLoadError;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import utilities.MockUtility;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.Assert.*;
+
+/**
+ * Class for testing SQLite Database.
+ *
+ * @author Rsl1122
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class SQLiteDatabaseNullSafetyTest extends SQLiteDatabaseTest {
```

Comment:
I think the database itself should at least check for null UUID. Currently if someone manages to drop in a null UUID it breaks some of the get methods due to `UUID.fromString(null)`

Reply:
After a quick check, the UUIDs used in the plugin internals seem to always be generated by `player.getUniqueId()`, so they cannot be null. The UUID can only be null when the database methods are called from the API layer. Putting the validation there rather than in SQLDatabaseManager allows to squeeze in a tiny bit of performance when the database methods are called by the plugin directly, as we don't need to perform validation on something we know will be correct, and avoids adding complexity to an already complicated class. 😉